src/components/workspace/qr-gen.tsx

```diff
@@ -0,0 +1,35 @@
+import { createEffect, createSignal, onCleanup } from "solid-js";
+import * as qrcode from "qrcode";
+
+export default function CertQrCode({ value }: { value: string }) {
+  const [qrCodeData, setQRCodeData] = createSignal<string | null>(null);
+
+  createEffect(() => {
+    if (value) {
+      const canvas = document.createElement("canvas");
```

Comment:
Please do not modify the DOM directly unless unavoidable. You may try [refs](https://www.solidjs.com/tutorial/bindings_refs). `createSignal` with a HTML element type also works.